src/Console/Application.php

```diff
@@ -187,43 +187,30 @@ private function parseConfiguration(InputInterface $input)
             $this->config = Config::fromFile($path);
         }
 
-        if ($input->hasParameterOption('--cli')) {
+        if ($input->getOption('cli')) {
             $this->config['adapter'] = 'cli';
         } elseif ($input->hasParameterOption('--fcgi')) {
             $this->config['adapter'] = 'fastcgi';
-            $this->config['fastcgiChroot'] = $input->getParameterOption('--fcgi-chroot');
-
-            if (!is_null($input->getParameterOption('--fcgi'))) {
-                $this->config['fastcgi'] = $input->getParameterOption('--fcgi');
-            }
+            $this->config['fastcgiChroot'] = $input->getOption('fcgi-chroot') ?? $this->config['fastcgiChroot'];
+            $this->config['fastcgi'] = $input->getOption('fcgi') ?? $this->config['fastcgi'];
         } elseif ($input->hasParameterOption('--web')) {
             $this->config['adapter'] = 'web';
             $this->config['webClient'] = $input->getOption('web') ?? 'FileGetContents';
-            $this->config['webPath'] = $input->getParameterOption('--web-path');
-            $this->config['webUrl'] = $input->getParameterOption('--web-url');
+            $this->config['webPath'] = $input->getOption('web-path') ?? $this->config['webPath'];
+            $this->config['webUrl'] = $input->getOption('web-url') ?? $this->config['webUrl'];
 
             if ($this->config['webClient'] === 'SymfonyHttpClient') {
-                if ($input->hasParameterOption('--web-allow-insecure')) {
-                    $this->config['webAllowInsecure'] = true;
-                }
-
-                if ($input->hasParameterOption('--web-basic-auth')) {
-                    $this->config['webBasicAuth'] = $input->getParameterOption('--web-basic-auth');
-                }
-
-                if ($input->hasParameterOption('--web-host')) {
-                    $this->config['webHost'] = $input->getParameterOption('--web-host');
-                }
+                $this->config['webAllowInsecure'] = $input->getOption('web-allow-insecure');
```

Comment:
I am not sure how to merge this here, I think this is a bit tricky since not providing it should set it to false, while providing it means true, no matter what the previous config value is. I don't think this is an issue so as previously this was not merged with the config from file either.

Reply:
I dont think its an issue as well.